### PR TITLE
main/opensmtpd: update to 6.0.2p1

### PR DESCRIPTION
**Pull request by Shizmob**
Removes old LibreSSL patch that seems to be checked properly now,
and fix some incorrect config.h checking. Also refactors the initd
script to be more natural/declarative.

### CODE REVIEW

File: main/opensmtpd/smtpd.initd [hunk 1]
```diff
@@ -2,23 +2,13 @@
 
 name="$SVCNAME"
 command="/usr/sbin/$SVCNAME"
-command_args="$SMTPD_OPTS"
+command_args="-F $SMTPD_OPTS"
+command_background=yes
+pidfile="/run/$SVCNAME.pid"
```
Comment (jirutka) [inline main/opensmtpd/smtpd.initd]: This variable should be named `$RC_SVCNAME` (see [openrc-run(8)](http://manpages.org/openrc-run/8)). And the same applies for `name="$SVCNAME"` above.

File: main/opensmtpd/smtpd.initd [hunk 2]
```diff
@@ -2,23 +2,13 @@
...
-start() {
-	ebegin "Starting smtpd ${CONF_MESSAGE}"
-	/usr/sbin/smtpd
-	eend $?
-}
-
-stop() {
-       ebegin "Stopping smtpd ${CONF_MESSAGE}"
-       /usr/sbin/smtpctl stop &> /dev/null
-       eend $?
```
Comment (jirutka) [inline main/opensmtpd/smtpd.initd]: 👍

File: main/opensmtpd/APKBUILD [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
-pkgver=5.9.2p1
-pkgrel=3
+pkgver=6.0.2p1
+pkgrel=0
 pkgdesc="secure, reliable, lean, and easy-to configure SMTP server"
 url="http://www.opensmtpd.org"
 arch="all"
 license="ISC"
-depends=""
+depends="libasr fts libevent db libressl"
```
Comment (jirutka) [inline main/opensmtpd/APKBUILD]: These dependencies are redundant, they are autodiscovered via dynamic linking by abuild:

```
>>> opensmtpd*: Tracing dependencies...
    busybox
    so:libasr.so.0
    so:libc.musl-x86_64.so.1
    so:libcrypto.so.38
    so:libdb-5.3.so
    so:libevent-2.1.so.6
    so:libfts.so.0
    so:libssl.so.39
    so:libz.so.1
```

Comment (Shizmob) [inline main/opensmtpd/APKBUILD]: I wasn't quite aware if that was simply to help the abuild developer -- good to know!

**Comment by Shizmob** [comment]
Updated!

**Pull request closed without merging**